indico/legacy/webinterface/rh/base.py

```diff
@@ -587,6 +587,15 @@ def process(self, params):
                 self._check_auth(params)
                 profile_name, res = self._do_process(profile)
                 signals.after_process.send()
+
+                if self.commit:
+                    # ensure we fail early (before sending out e-mails)
+                    # in case there are DB constraint violations, etc...
+                    db.session.flush()
```

Comment:
You could also call `db.enforce_constraints()` instead - this already flushes but also ensures that e.g. inconsistent-timetable errors are triggered